sdk/main/src/ED25519PrivateKey.cc

```diff
@@ -157,12 +158,98 @@ std::vector<unsigned char> ED25519PrivateKey::toBytes() const
     std::cout << "ED25519PrivateKey toBytes I2D error" << std::endl;
   }
 
-  return outputBytes;
+  // don't return the algorithm identification bytes
+  return { outputBytes.begin() + 16, outputBytes.end() };
 }
 
 EVP_PKEY* ED25519PrivateKey::bytesToPKEY(const std::vector<unsigned char>& keyBytes)
 {
-  const unsigned char* rawKeyBytes = &keyBytes.front();
-  return d2i_PrivateKey(EVP_PKEY_ED25519, nullptr, &rawKeyBytes, (long)keyBytes.size());
+  std::vector<unsigned char> fullKeyBytes;
+  // If there are only 32 key bytes, we need to add the algorithm identifier bytes, so that OpenSSL can correctly decode
+  if (keyBytes.size() == 32)
+  {
+    fullKeyBytes = prependAlgorithmIdentifier(keyBytes);
+  }
+  else
+  {
+    fullKeyBytes = keyBytes;
+  }
+
+  const unsigned char* rawKeyBytes = &fullKeyBytes.front();
+
+  return d2i_PrivateKey(EVP_PKEY_ED25519, nullptr, &rawKeyBytes, (long)fullKeyBytes.size());
+}
+
+std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::fromBIP39Mnemonic(const MnemonicBIP39& mnemonic,
+                                                                        const std::string& passphrase)
+{
+  return fromSeed(mnemonic.toSeed(passphrase));
+}
+
+std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::fromHMACOutput(const std::vector<unsigned char>& hmacOutput)
+{
+  if (hmacOutput.size() != 64) {
+    throw std::runtime_error("Unexpected size for hmac output");
+  }
+
+  // the first 32 bytes of the hmac are the new key material. the algorithm identifier must come first, though
+  std::vector<unsigned char> fullKey = prependAlgorithmIdentifier({ hmacOutput.begin(), hmacOutput.begin() + 32 });
+
+  // chain code is the next 32 bytes of the computed hmac
+  std::vector<unsigned char> chainCode(hmacOutput.begin() + 32, hmacOutput.end());
+
+  return std::unique_ptr<ED25519PrivateKey>(new ED25519PrivateKey(bytesToPKEY(fullKey), chainCode));
+}
+
+std::unique_ptr<ED25519PrivateKey> ED25519PrivateKey::fromSeed(const std::vector<unsigned char>& seed)
+{
+  std::string keyString = "ed25519 seed"; // as defined by SLIP 0010
```

Comment:
Since it's only used in 1 place, would you be fine keeping it inline, but making it static const? I'm not sure if this is generally considered good or not

```
static const std::string keyString = "ed25519 seed"; // as defined by SLIP 0010
```